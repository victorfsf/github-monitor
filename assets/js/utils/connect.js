import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

const connectRouter = (props, component) => (
  withRouter(connect(props)(component))
);

export default connectRouter;
