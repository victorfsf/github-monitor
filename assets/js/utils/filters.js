
const filterDuplicates = (l, key) => (
  l.filter(
    (el, index, self) => self.findIndex(
      t => ((t[key] === el[key])),
    ) === index,
  )
);


export default filterDuplicates;
