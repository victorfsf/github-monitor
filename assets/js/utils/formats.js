import moment from 'moment';


const filterDuplicates = (l, key) => (
  l.filter(
    (el, index, self) => self.findIndex(
      t => ((t[key] === el[key])),
    ) === index,
  )
);


const formatDate = date => (
  moment(date).format('MMM, D YYYY [at] h:mm:ss A')
);


export {
  filterDuplicates,
  formatDate,
};
