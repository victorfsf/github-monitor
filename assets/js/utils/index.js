import filterDuplicates from './filters';
import Urls from './urls';
import connectRouter from './connect';


export {
  Urls,
  connectRouter,
  filterDuplicates,
};
