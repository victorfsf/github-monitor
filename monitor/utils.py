import hmac
from hashlib import sha1

from django.conf import settings

from monitor.models import Author


def hub_signature_verify(value, expected):

    if not value or not expected:
        return False

    calc = hmac.new(settings.HUB_SECRET.encode(), value, sha1).hexdigest()
    return hmac.compare_digest(f'sha1={calc}', expected)


def get_author(payload):
    github_id = payload.get('id') or payload.get('github_id')
    email = payload.get('email')
    kwargs = {
        'github_id': github_id
    } if github_id else {
        'email': email
    }
    author, _ = Author.objects.get_or_create(**kwargs)
    fields = {
        'github_id': github_id,
        'email': email,
        'name': payload.get('name'),
        'login': payload.get('login')
    }
    for field, value in fields.items():
        if value:
            setattr(author, field, value)
    author.save()
    return author
