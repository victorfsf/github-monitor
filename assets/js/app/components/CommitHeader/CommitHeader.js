import React from 'react';
import PropTypes from 'prop-types';
import Box from 'app/components/Box';
import { formatDate } from 'utils';


const CommitHeader = props => (
  <Box>
    <div className="align-middle">
      <div className="row">
        <div className="col-sm-12 col-md-6">
          Captured Commits {props.extraHeader}
          <button
            type="button"
            className="btn btn-sm btn-link"
            onClick={props.onClick}
          >
            <i className="fa fa-refresh" />
          </button>
        </div>
        <div className="col-sm-12 col-md-6">
          <small className="float-md-right">
            Count: {props.count} Commits
          </small>
        </div>
      </div>
      <div className="row">
        <div className="col-12">
          <small className="text-muted">
            Last Update: {
              props.lastUpdate ?
              formatDate(props.lastUpdate) :
              '...'
            }
          </small>
        </div>
      </div>
    </div>
  </Box>
  );


CommitHeader.propTypes = {
  onClick: PropTypes.func.isRequired,
  lastUpdate: PropTypes.string,
  extraHeader: PropTypes.string,
  count: PropTypes.number,
};


CommitHeader.defaultProps = {
  extraHeader: '',
  lastUpdate: null,
  count: 0,
};


export default CommitHeader;
