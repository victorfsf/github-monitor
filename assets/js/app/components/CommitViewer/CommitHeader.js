import React from 'react';
import PropTypes from 'prop-types';
import Box from 'app/components/Box';
import { GITHUB_URL } from 'constants/github';
import { formatDate } from 'utils';


const CommitHeader = props => (
  <Box>
    <div className="align-middle">
      <div className="row">
        <div className="col-sm-12 col-md-6">
          Captured Commits
          <button
            type="button"
            className="btn btn-sm btn-link"
            onClick={props.onClick}
          >
            <i className="fa fa-refresh" />
          </button>
          {props.repo && (
            <small>
              <a href={`${GITHUB_URL}${props.repo}`} target="blank_">
                {props.repo}
              </a>
            </small>
          )}
        </div>
        <div className="col-sm-12 col-md-6">
          <small className="float-md-right">
            {props.count} Commits
          </small>
        </div>
      </div>
      <div className="row">
        <div className="col-12">
          <small className="text-muted">
            Last Update: {
              props.lastUpdate ?
              formatDate(props.lastUpdate) :
              'Loading...'
            }
          </small>
        </div>
      </div>
    </div>
  </Box>
  );


CommitHeader.propTypes = {
  onClick: PropTypes.func.isRequired,
  lastUpdate: PropTypes.string,
  repo: PropTypes.string,
  count: PropTypes.number,
};


CommitHeader.defaultProps = {
  repo: '',
  lastUpdate: null,
  count: 0,
};


export default CommitHeader;
