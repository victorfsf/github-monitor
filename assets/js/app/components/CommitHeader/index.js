import CommitHeader from './CommitHeader';

export default CommitHeader;
