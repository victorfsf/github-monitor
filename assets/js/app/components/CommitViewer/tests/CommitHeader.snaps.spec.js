import React from 'react';
import renderer from 'react-test-renderer';
import CommitHeader from 'app/components/CommitViewer/CommitHeader';
import * as utils from 'utils';

utils.formatDate = jest.fn(() => 'MMM D, YYYY [at] h:mm:ss A');


describe('CommitHeader', () => {
  let Component;
  let tree;

  test('renders', () => {
    Component = renderer.create(
      <CommitHeader repo="test/test" onClick={e => e} />,
    );
    tree = Component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders with lastUpdate and count', () => {
    Component = renderer.create(
      <CommitHeader
        count={10}
        repo="test/test"
        onClick={e => e}
        lastUpdate="today"
      />,
    );
    tree = Component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('does not render without onClick', () => {
    global.console = {
      error: jest.fn(),
    };
    renderer.create(<CommitHeader />);
    expect(global.console.error).toHaveBeenCalledTimes(1);
  });
});
