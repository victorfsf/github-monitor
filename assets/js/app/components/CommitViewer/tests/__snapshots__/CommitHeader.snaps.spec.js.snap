// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommitHeader renders 1`] = `
<div
  className="my-3 p-3 bg-white rounded box-border"
>
  <div
    className="align-middle"
  >
    <div
      className="row"
    >
      <div
        className="col-sm-12 col-md-8"
      >
        Captured Commits
        <button
          className="btn btn-sm btn-link"
          onClick={[Function]}
          type="button"
        >
          <i
            className="fa fa-refresh"
          />
        </button>
        <small>
          <a
            href="https://github.com/test/test"
            target="blank_"
          >
            test/test
          </a>
        </small>
      </div>
      <div
        className="col-sm-12 col-md-4"
      >
        <small
          className="float-md-right"
        >
          0
           Commits
        </small>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <small
          className="text-muted"
        >
          Last Update:
          <span
            className="ml-1"
          >
            Loading...
          </span>
        </small>
      </div>
    </div>
  </div>
</div>
`;

exports[`CommitHeader renders with lastUpdate and count 1`] = `
<div
  className="my-3 p-3 bg-white rounded box-border"
>
  <div
    className="align-middle"
  >
    <div
      className="row"
    >
      <div
        className="col-sm-12 col-md-8"
      >
        Captured Commits
        <button
          className="btn btn-sm btn-link"
          onClick={[Function]}
          type="button"
        >
          <i
            className="fa fa-refresh"
          />
        </button>
        <small>
          <a
            href="https://github.com/test/test"
            target="blank_"
          >
            test/test
          </a>
        </small>
      </div>
      <div
        className="col-sm-12 col-md-4"
      >
        <small
          className="float-md-right"
        >
          10
           Commits
        </small>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <small
          className="text-muted"
        >
          Last Update:
          <span
            className="ml-1"
          >
            MMM D, YYYY [at] h:mm:ss A
          </span>
        </small>
      </div>
    </div>
  </div>
</div>
`;
